javascript/code-challenges/challenges-09.test.js

```diff
@@ -0,0 +1,357 @@
+'use strict';
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 1 - Review
+
+Write a function that finds the maximum value in an array
+using the 'reduce' method.
+
+E.g. [4,2,7,5,9,2] -> 9
+------------------------------------------------------------------------------------------------ */
+const maxInArray = (arr) => arr.reduce((acc, val) => acc > val ? acc : val);
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named getCourseKeys that takes in the courseInfo object and returns an array containing the keys for the courseInfo object.
+
+For example: (['name', 'duration', 'topics', 'finalExam']).
+------------------------------------------------------------------------------------------------ */
+const courseInfo = {
+  name: 'Code 301', duration: { dayTrack: '4 weeks', eveningTrack: '8 weeks' },
+  topics: ['SMACSS', 'APIs', 'NodeJS', 'SQL', 'jQuery', 'functional programming'],
+  finalExam: true
+};
+
+const getCourseKeys = (obj) => Object.keys(obj);
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named checkValues that takes in an object and a value and returns true if the value is in the object.
+
+
+------------------------------------------------------------------------------------------------ */
+
+const checkValues = (obj, value) => Object.values(obj).includes(value);
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+You are given an object with names and their coresponding phone numbers that looks like this:
+{
+  'Grace Hopper': '222-303-5938',
+  'Ada Lovelace': '222-349-9842',
+  'Alan Turing': '222-853-5933'
+}
+
+HR has asked you to change the data to make it easier to print so that it looks like this:
+[
+  'Grace Hopper: 222-303-5938',
+  'Ada Lovelace: 222-349-9842',
+  'Alan Turing: 222-853-5933'
+]
+
+------------------------------------------------------------------------------------------------ */
+
+const updateNumbers = (obj) => {
+  let keys = Object.keys(obj);
+  let values = Object.values(obj);
+  const newArr = [];
+  for (let i = 0; i < keys.length; i++) {
+    newArr.push(keys[i].concat(': ', values[i]));
```

Comment:
using .map( ) could really shorten up your answer here, but wouldn't really make it any more efficient so, your answer works totally fine.